rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -127,6 +153,11 @@ get_node_names(
       if (ns_found != map.end()) {
         namespace_ = std::string(ns_found->second.begin(), ns_found->second.end());
       }
+
+      if (security_context_found != map.end()) {
+        security_context = std::string(
+          security_context_found->second.begin(), security_context_found->second.end());
```

Comment:
@ivanpauno this is a peculiar way of copying an `std::string` and I suspect less efficient than a plain copy-assignment.

Reply:
See https://github.com/ros2/rmw/blob/f76511333709a6f5d1e4543129ec73f2cd2b4f06/rmw/include/rmw/impl/cpp/key_value.hpp#L31.

Reply:
That's a peculiar API. I wonder why a sequence of bytes. Is it to support other encodings in the future? cc @dirk-thomas